libres/lib/res_util/block_fs.cpp

```diff
@@ -564,7 +431,7 @@ static bool block_fs_fseek_valid_node(block_fs_type *block_fs) {
     int status;
     while (true) {
         if (fread(&byte, sizeof byte, 1, block_fs->data_stream) == 1) {
-            if (byte == NODE_IN_USE_BYTE || byte == NODE_FREE_BYTE) {
+            if (byte == NODE_IN_USE_BYTE) {
```

Comment:
Can we be a little smarter (+ make the code a little clearer) here...  reading first a single byte, then perhaps an integer seems a little over the top?

Reply:
Not sure what you mean. Are you suggesting we don't read the byte but just attempt to read the int to begin with?

Reply:
I assume this file can be addressed byte-wise, hence we cannot just read and check the full int. But instead of reading one byte, then fseek back one byte and read the int, we might read some reasonable-sized block into memory, scan it for `NODE_IN_USE` and fseek() to the right pos if required. Would probably be quite a lot faster, but perhaps more suitable for a different PR...